File: server/generate_pdf.ts
import fs from 'fs';
import { PDFDocument, StandardFonts, rgb } from 'pdf-lib';
import path from 'path';

async function createPdf() {
    const pdfDoc = await PDFDocument.create();
    const timesRomanFont = await pdfDoc.embedFont(StandardFonts.TimesRoman);

    const page = pdfDoc.addPage();
    const { width, height } = page.getSize();
    const fontSize = 16;

    page.drawText('Sample English Complaint Document', {
        x: 50,
        y: height - 4 * fontSize,
        size: fontSize,
        font: timesRomanFont,
        color: rgb(0, 0, 0),
    });

    page.drawText('Plaintiff: John Doe', { x: 50, y: height - 6 * fontSize, size: 12, font: timesRomanFont });
    page.drawText('Defendant: Acme Corp', { x: 50, y: height - 7 * fontSize, size: 12, font: timesRomanFont });
    page.drawText('Disputed Amount: 50,000 USD', { x: 50, y: height - 8 * fontSize, size: 12, font: timesRomanFont });
    page.drawText('Cause of action: Breach of Contract due to delayed delivery of goods.', { x: 50, y: height - 9 * fontSize, size: 12, font: timesRomanFont });

    const pdfBytes = await pdfDoc.save();
    fs.writeFileSync(path.join(__dirname, 'test_complaint.pdf'), pdfBytes);
    console.log("PDF generated successfully.");
}

createPdf();

File: server/generate_pdf.py
from reportlab.pdfgen import canvas
from reportlab.pdfbase import pdfmetrics
from reportlab.pdfbase.ttfonts import TTFont
import sys

def create_pdf(text_file, output_pdf):
    c = canvas.Canvas(output_pdf)
    
    # We will just write basic ASCII text to avoid font issues to test parsing
    text = "Complaint. Plaintiff: John Doe. Defendant: Acme Corp. Disputed amount: 50,000 USD. Breach of contract. The plaintiff asks the court to award damages."
    
    y = 800
    for line in text.split(". "):
        c.drawString(100, y, line)
        y -= 20
        
    c.save()

if __name__ == "__main__":
    create_pdf("test_complaint.txt", "test_complaint.pdf")
